src/main/java/com/google/devtools/build/lib/runtime/BlazeRuntime.java

```diff
@@ -1275,7 +1275,9 @@ private static BlazeRuntime newRuntime(
     PathFragment outputBase = startupOptions.outputBase;
     PathFragment realExecRootBase = outputBase.getRelative(ServerDirectories.EXECROOT);
 
-    maybeForceJNIByGettingPid(installBase); // Must be before first use of JNI.
+    // Force JNI linking before the first real use of JNI to emit a helpful error message now that
+    // we have the install base path handy.
+    forceJniLinking(installBase);
```

Comment:
Since we now throw unconditionally here if the JNI can't be loaded, do we still need the fallback to JavaIoFileSystem in BazelFileSystemModule? Could we just handle the Unix and Windows cases similarly there?

Reply:
That wasn't intentional, I added back the availability check.